@here/harp-mapview-decoder/lib/TileLoader.ts

```diff
@@ -148,6 +149,17 @@ export class TileLoader {
                     this.requestController = undefined;
                 }
                 break;
+
+            case TileLoaderState.Canceled:
```

Comment:
Tentative.
This just adds bytes that add no value for final application (we have worse boilerplate here and there but here we add more).

`cancel` was written as idempotent method, which is very useful property - you expect only certain effect, without looking at current state.

It's no error to `cancel` loader that for some particular reason is in `loaded` state (or any other), it's should be just  `no-op`.